aea/protocols/dialogue/base.py

```diff
@@ -1134,18 +1134,50 @@ def remove(self, dialogue_label: DialogueLabel) -> None:
 
         :param dialogue_label: label of the dialogue to remove
         """
-        dialogue = self._dialogues_by_dialogue_label.pop(dialogue_label, None)
+        dialogue_reference = dialogue_label.dialogue_reference
 
-        self._incomplete_to_complete_dialogue_labels.pop(dialogue_label, None)
+        # Check if dialogue_label is a complete label
+        is_complete_label = (
+            dialogue_reference[0] != Dialogue.UNASSIGNED_DIALOGUE_REFERENCE
+            and dialogue_reference[1] != Dialogue.UNASSIGNED_DIALOGUE_REFERENCE
+        )
 
-        if dialogue_label in self._terminal_state_dialogues_labels:
-            self._terminal_state_dialogues_labels.remove(dialogue_label)
+        complete_dialogue_label = dialogue_label
```

Comment:
```suggestion
        incomplete_dialogue_label = dialogue_label
```
default path is with is_complete_label = False below

Reply:
Make sure below you are very explicit when using complete/incomplete or just dialogue_label...